sources/packages/backend/apps/queue-consumers/src/processors/schedulers/sfas-integration/_tests_/sfas-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,191 @@
+      "Should not insert a legacy restriction if a legacy restriction is already active in SIMS for the student, " +
+      "Should insert a restriction after mapping it to SIMS restriction if it is either inactive or is not at all present in the SIMS student restrictions, " +
+      "Should not insert a restriction if it has a restriction resolution date.",
+    async () => {
+      const studentExists = await db.student.findOne({
+        where: {
+          birthDate: getISODateOnlyString(new Date("1998-03-24")),
+          user: { lastName: "FOUR" },
+          sinValidation: { sin: "900041310" },
+        },
+      });
+      let student = studentExists;
+      if (!studentExists) {
+        student = await saveFakeStudent(db.dataSource);
+        // Update the student to ensure that the student imported from SFAS is the same student as the one created above.
+        student.birthDate = getISODateOnlyString(new Date("1998-03-24"));
+        student.user.lastName = "FOUR";
+        student.sinValidation.sin = "900041310";
+        await db.student.save(student);
+      }
+      const restrictionB6B = await db.restriction.findOne({
+        where: { restrictionCode: "B6B" },
+      });
+      const restrictionLGCY = await db.restriction.findOne({
+        where: { restrictionCode: "LGCY" },
+      });
+      const restrictionSSR = await db.restriction.findOne({
+        where: { restrictionCode: "SSR" },
+      });
+      await saveFakeStudentRestriction(db.dataSource, {
+        student,
+        restriction: restrictionB6B,
+      });
+      await saveFakeStudentRestriction(db.dataSource, {
+        student,
+        restriction: restrictionLGCY,
+      });
+      const studentRestrictionSSR = await saveFakeStudentRestriction(
+        db.dataSource,
+        {
+          student,
+          restriction: restrictionSSR,
+        },
+      );
+      studentRestrictionSSR.isActive = false;
+      await db.studentRestriction.save(studentRestrictionSSR);
+      // Queued job.
+      const job = createMock<Job<void>>();
+      mockDownloadFiles(sftpClientMock, [SFAS_ALL_RESTRICTIONS_FILENAME]);
+      // Act
+      await processor.processSFASIntegrationFiles(job);
+      // Assert
+      // Expect the file was deleted from SFTP.
+      expect(sftpClientMock.delete).toHaveBeenCalled();
+      // Expect a total of 5 restrictions to be inserted.
+      // Two originally inserted active restrictions (LGCY & B6B) from before the file processing and then two AF restrictions and one SSR added from the file import.
+      const studentRestrictions = await db.studentRestriction.find({
+        select: {
+          id: true,
+          restriction: {
+            restrictionCode: true,
+          },
+        },
+        relations: { restriction: true },
+        where: { student: { id: student.id }, isActive: true },
+      });
+      expect(studentRestrictions.length).toBe(5);
+      const restrictionsMap = new Map<RestrictionCode, number>();
```

Comment:
for any restrictions we want to check we can assert in 2 line like this.


![image](https://github.com/bcgov/SIMS/assets/54600590/9cfb86bb-ae4d-40b0-957c-83114f569c52)

Reply:
I think Map is more concise. Otherwise, for each restriction, we will be running the filter method again and again which I believe is more expensive than using a `Map` and finding the count of each restriction in one run.

Reply:
@andrewsignori-aot @lewischen-aot @andrepestana-aot @guru-aot Need some advice here on which approach to follow. Any suggestions?

Reply:
What if the below?
```ts
const restrictionsCount: Record<string, number> = {};
studentRestrictions.forEach((restriction) => {
  const code = restriction.restriction.restrictionCode;
  restrictionsCount[code] = (restrictionsCount[code] || 0) + 1;
});
expect(restrictionsCount).toBe({
  [RestrictionCode.LGCY]: 1,
  [RestrictionCode.AF]: 2,
  [RestrictionCode.B6B]: 1,
  [RestrictionCode.SSR]: 1,
});
```

Reply:
`Record` or `Map`, fine with either.